graph/service.go

```diff
@@ -29,8 +30,22 @@ func (s *TagStore) Lookup(name string) (*types.ImageInspect, error) {
 		return nil, fmt.Errorf("No such image: %s", name)
 	}
 
+	var tags = make([]string, 0)
+
+	s.Lock()
+	for repoName, repository := range s.Repositories {
+		for ref, id := range repository {
+			if id == image.ID {
+				imgRef := utils.ImageReference(repoName, ref)
+				tags = append(tags, imgRef)
+			}
+		}
+	}
+	s.Unlock()
```

Comment:
If we need a lock at all, can we at least make it a read-lock so we don't block all interactions on the image DB?